### DISCUSSION TITLE
Support for Buderus Ecomatic HS4201 (Remote FMEC)

### DISCUSSION THREAD

**Discussion by Technikfreak72** [emsesp/EMS-ESP32#1789] [Q&A]
I have an Ecomatic HS4201 with a display remote FMEC. You can control everything with the remote an it also messures the temperature of the room in which it is installed. The communication is wired over two wires and the remote also needs no other power supply. So I think it is a basic EMS bus. I have a wired Tado thermostate which is configured as EMS and it seems to work but I am not 100% sure sice Tado do not show any connection status or log.
When I try to connect via EMS-ESP by replacing the remote I could not get a connection.
Could anybody help me?

**Comment by proddy** [comment]
do you see anything at all in the EMS-ESP log when its set to ALL or DEBUG? Look for incoming Rx traffic. Then play with the settings. https://emsesp.github.io/docs/Troubleshooting/#many-rx-errors

**Reply by Technikfreak72** [reply]
No, I think it has a different baud rate. Now I know it is a different bus type from Buderus called BF-Bus. The Buderus Ecomatic was a top selling product and many are still working as mine after 30 years! To make it smart  would be fantastic.
Is there a simple way to test different baud rates and record the traffic on the bus?

**Reply by proddy** [reply]
EMS is 9600 baud. If you're comfortable with coding, you can download the EMS-ESP code, change the baud in `emsuart_esp32.h`, build and test it out.

**Reply by Technikfreak72** [reply]
Thank you

**Comment by GernotAlthammer** [comment]
Hello,
are you still looking for a solution to control your Ecomatic 4000 HS4201 heater system?

This is my solution:
https://github.com/GernotAlthammer/buderus_ecomatic4000

Regards
